Automattic/pocket-casts-android
# Continue to upgrade screen from google sign in when appropriate

## Description
This fixes the onboarding flow when someone clicks an upgrade prompt and then logs into a free account using Google. See #1184 for more info.

Fixes #1184 

## Testing Instructions

If you test this on a debug build, note that tapping on the "Upgrade to Plus" button can freeze the app if you're already logged in (https://github.com/Automattic/pocket-casts-android/issues/1158). Also note that the upgrade screen won't have any subscription options unless you make the changes needed to pull in subscriptions (which are not needed to test this PR).

### 1. Fixes upgrade flow
1. Fresh install of the app
2. Go to `Profile` tab → ⚙️ → Pocket Casts Plus
3. Tap on "Upgrade to Plus"
4. Tap "Continue with Google"
5. Log into an already existing free Pocket Casts account
6. ✅ Observe that you are taken to the onboarding flow's upgrade screen.

### 2. Check for regressions

#### 2a. Upgrading and creating a new Pocket Casts account with Google sign in
1. Fresh install of the app
2. Go to `Profile` tab → ⚙️ → Pocket Casts Plus
3. Tap on "Upgrade to Plus"
4. Tap "Continue with Google"
5. Create a new Pocket Casts account (i.e., select a Google login that is not associated with a PC account already)
6. ✅ Observe that you are taken to the recommendations screen in the onboarding flow.

#### 2b. Signing into a free account using Google outside the upgrade flow
1. Fresh install of the app
2. Tap "Continue with Google"
3. Create a new Pocket Casts account (i.e., select a Google login that is not associated with a PC account already)
4. ✅ Observe that you are taken to the recommendations screen in the onboarding flow.

#### 2c. Creating a new Google account in the regular onboarding flow
1. Fresh install of the app
2. Tap "Continue with Google"
3. Log into an already existing free Pocket Casts account
4. ✅ Observe that you are taken out of the onboarding flow and into the app

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/4656348/ba087724-f960-4bfd-89ea-84f386f3a29f

## Checklist
